testing/src/cypress/e2e/02-institution/04-manage-location.cy.ts

```diff
@@ -1,121 +1,518 @@
-import InstitutionHelperActions from "./common-helper-functions.cy";
+import InstitutionHelperActions from "../../custom-command/institution/common-helper-functions.cy";
+import data from "../data/institution/manage-location.json";
 
 const dashboardInstitutionObject = new DashboardInstitutionObject();
 const institutionManageLocationObject = new ManageLocationObject();
 const institutionHelperActions = new InstitutionHelperActions();
 
-function manageInstitution() {
+function loginAndClickOnManageInstitution() {
+  institutionHelperActions.loginIntoInstitutionSingleLocation();
   dashboardInstitutionObject.dashboardButton().click();
   dashboardInstitutionObject.manageInstitutionButton().click();
 }
 
-function intercept() {
+function loginAndClickOnEditLocation() {
+  loginAndClickOnManageInstitution();
   cy.intercept("GET", "**/location/**").as("location");
-}
-
-function institutionCode() {
-  intercept();
-  manageInstitution();
   institutionManageLocationObject.manageLocationButton().click();
   institutionManageLocationObject.editLocationButton().click();
   cy.wait("@location");
-  institutionManageLocationObject.institutionCode().should("be.disabled");
 }
 
-function institutionInputText() {
-  institutionCode();
-  institutionManageLocationObject.locationName().should("be.disabled");
-  institutionManageLocationObject.addressFirst().should("be.disabled");
-  institutionManageLocationObject.cityInputText().should("be.disabled");
-  institutionManageLocationObject.countryInputText().should("be.disabled");
+function loginAndClickOnAddNewLocation() {
+  loginAndClickOnManageInstitution();
+  institutionManageLocationObject.addNewLocationButton().click();
+}
+
+function verifyLocationAndContactDetails() {
+  institutionManageLocationObject.institutionCode().should("be.visible");
+  institutionManageLocationObject.address1Text().should("be.visible");
+  institutionManageLocationObject.address2Text().should("be.visible");
+  institutionManageLocationObject.countryText().should("be.visible");
+  institutionManageLocationObject.cityText().should("be.visible");
+  institutionManageLocationObject.primaryContactText().should("be.visible");
+  institutionManageLocationObject.firstNameText().should("be.visible");
+  institutionManageLocationObject.lastNameText().should("be.visible");
+  institutionManageLocationObject.phoneNumberText().should("be.visible");
+  institutionManageLocationObject.emailText().should("be.visible");
+}
+
+function verifyInvalidEmailFormatsAreNotAllowed() {
+  data.invalidData.emailAddress.forEach((text: string) => {
+    institutionManageLocationObject.emailInputText().clear().type(text);
+    cy.contains("Email must be a valid email.").should("be.visible");
+  });
+}
+
+function verifyValidEmailFormatsAreAllowed() {
+  data.validData.emailAddress.forEach((text: string) => {
+    institutionManageLocationObject.emailInputText().clear().type(text);
```

Comment:
mm, why 2 email validations here? `verifyInvalidEmailFormatsAreNotAllowed` and `verifyValidEmailFormatsAreAllowed`

Reply:
we need to validate 
- Invalid formats are not allowed and 
- valid email formats are allowed. 
So separated them out